content/docs/getting-started/hello-world.md

```diff
@@ -71,93 +56,74 @@ Next, we'll add some to the `client` `TcpStream`. This asynchronous task now cre
 the stream and then yields it once it's been created for additional processing.
 
 ```rust
-# #![deny(deprecated)]
-# extern crate tokio;
-#
-# use tokio::net::TcpStream;
-# use tokio::prelude::*;
-# fn main() {
-# let addr = "127.0.0.1:6142".parse().unwrap();
-let client = TcpStream::connect(&addr).and_then(|stream| {
-    println!("created stream");
-
-    // Process stream here.
-
-    Ok(())
-})
-.map_err(|err| {
-    // All tasks must have an `Error` type of `()`. This forces error
-    // handling and helps avoid silencing failures.
-    //
-    // In our example, we are only going to log the error to STDOUT.
-    println!("connection error = {:?}", err);
-});
-# }
+use tokio::net::TcpStream;
+
+fn main() {
+    let addr = "127.0.0.1:6142";
+    let client = TcpStream::connect(addr);
+
+    let future = async move {
+        match client.await {
+            Ok(_stream) => {
+                println!("created stream");
+
+                // Process stream here.
+            }
+            Err(err) => {
+                // In our example, we are only going to log the error to STDOUT.
+                println!("connection error = {:?}", err);
+            }
+        };
+    };
+}
 ```
 
-The call to `TcpStream::connect` returns a [`Future`] of the created TCP stream.
+Inside the block the call to `TcpStream::connect` returns a [`Future`] of the created TCP stream.
 We'll learn more about [`Futures`] later in the guide, but for now you can think of
 a [`Future`] as a value that represents something that will eventually happen in the
 future (in this case the stream will be created). This means that `TcpStream::connect` does
 not wait for the stream to be created before it returns. Rather it returns immediately
-with a value representing the work of creating a TCP stream. We'll see down below when this work
-_actually_ gets executed.
-
-The `and_then` method yields the stream once it has been created. `and_then` is an
-example of a combinator function that defines how asynchronous work will be processed.
-
-Each combinator function takes ownership of necessary state as well as the
-callback to perform and returns a new `Future` that has the additional "step"
-sequenced. A `Future` is a value representing some computation that will complete at
-some point in the future.
+with a value representing the work of creating a TCP stream.
 
-It's worth reiterating that returned futures are lazy, i.e., no work is performed when
-calling the combinator. Instead, once all the asynchronous steps are sequenced, the
-final `Future` (representing the entire task) is 'spawned' (i.e., run). This is when
-the previously defined work starts getting run. In other words, the code we've written
-so far does not actually create a TCP stream.
+Note the `async`-block that follows. Contents of the async block are not executed immediately but are rather
```

Comment:
Some minor nits:
```suggestion
Note the `async` block that follows. Contents of the `async` block are not executed immediately, but are rather
```

Reply:
Also, we use "rather" in the last sentence of the previous paragraph. What about something like:
```suggestion
Note the `async` block that follows. Like the `TcpStream::connect` future, the contents of the `async` block are not executed immediately. Instead, they are
```